server_utils/origin.go

```diff
@@ -758,11 +784,29 @@ func CheckOriginSentinelLocations(exports []OriginExport) (ok bool, err error) {
 			if path.Base(sentinelPath) != sentinelPath {
 				return false, errors.Errorf("invalid SentinelLocation path for StoragePrefix %s, file must not contain a directory. Got %s", export.StoragePrefix, export.SentinelLocation)
 			}
-			fullPath := filepath.Join(export.StoragePrefix, sentinelPath)
-			_, err := os.Stat(fullPath)
+
+			fullPath := filepath.Join(export.FederationPrefix, sentinelPath)
+			tkn, err := generateFileTestScitoken(sentinelPath)
+			if err != nil {
+				return false, errors.Wrap(err, "Failed to generate self-auth token for sentinel file check")
```

Comment:
"file" --> "object"

Another note is that I think we want the wrapped error to begin with a lowercase "f". The error message here gets concatenated with other error strings as it bubbles back up through the stack, so the golang convention is to start error strings without capitalization. This comment applies to other error messages in this section, too.